ouroboros-consensus/src/Ouroboros/Storage/VolatileDB/Util.hs

```diff
@@ -75,32 +78,23 @@ wrapFsError :: Monad m
 wrapFsError fsErr volDBErr action =
     tryVolDB fsErr volDBErr action >>= either (throwError volDBErr) return
 
--- | Parses the 'FsPath' of each filename and zips them together.
--- If any 'FsPath' does not parse, we return an error.
+-- | Parses the 'FileId' of each 'FsPath' and zips them together.
+-- If any parsing does not parse, we abort with an error.
 parseAllFds :: forall blockId.
                [FsPath]
             -> Either (VolatileDBError blockId) [(FileId, FsPath)]
-parseAllFds = mapM go
-  where
-    go :: FsPath -> Either (VolatileDBError blockId) (FileId, FsPath)
-    go file = case parseFd file of
-      Nothing -> Left $ UnexpectedError $ ParserError $ InvalidFilename file
-      Just fd -> Right (fd, file)
+parseAllFds = mapM $ \f -> (,f) <$> parseFd f
 
--- | Throws an error if one of the given file names does not parse.
+-- | If any 'FsPath' does not parse, we return an error.
 findLastFd :: forall blockId.
               Set FsPath
            -> Either (VolatileDBError blockId) (Maybe FileId)
-findLastFd = foldM go Nothing
-  where
-    maxMaybe :: Ord a => Maybe a -> a -> a
-    maxMaybe ma a = case ma of
-      Nothing -> a
-      Just a' -> max a' a
-    go :: Maybe FileId -> FsPath -> Either (VolatileDBError blockId) (Maybe FileId)
-    go fd file = case parseFd file of
-      Nothing  -> Left $ UnexpectedError $ ParserError $ InvalidFilename file
-      Just fd' -> Right $ Just $ maxMaybe fd fd'
+findLastFd = fmap safeMaximum . mapM parseFd . Set.toList
+
+maxMaybe :: Ord a => Maybe a -> a -> a
```

Comment:
Is this function still needed? I'd like to get rid of it. Use `safeMaximum` instead.